packages/web-app/lib/connectors.tsx

```diff
@@ -33,14 +33,23 @@ const gnosis: Chain = {
 };
 
 // Returns and alchemy provider based on the chain
-// if the chain is goerli, use the goerly alchemy api key from the env
-// else, use the mainnet alchemy api key from the env
 const customAlchemyProvider = ({ priority, stallTimeout, weight }: FallbackProviderConfig) => {
 	return (chain: Chain) => {
-		const apiKey =
-			chain.id === 5
-				? process.env.NEXT_PUBLIC_ALCHEMY_API_KEY_GOERLI
-				: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY;
+		let apiKey: any;
+		switch (chain?.id) {
+			case 1:
+				apiKey = process.env.NEXT_PUBLIC_ALCHEMY_API_KEY;
+				break;
+			case 5:
+				apiKey = process.env.NEXT_PUBLIC_ALCHEMY_API_KEY_GOERLI;
+				break;
+			case 11155111:
+				apiKey = process.env.NEXT_PUBLIC_ALCHEMY_API_KEY_SEPOLIA;
```

Comment:
Does this mean that I have to add this variable in the Vercel config?

Reply:
I followed the old code, but it would be better if we only needed NEXT_PUBLIC_ALCHEMY_API_KEY.

Reply:
Yes, I think so also.  Having one variable name per chain is just weird.  Let's refactor those later, in a separate pull request:  https://github.com/nation3/agreements-app/issues/139

Reply:
I already update it in my latest commit

Reply:
Hey, wait a second, something is weird here. We don't need NEXT_PUBLIC_CHAIN since our app supports multiple chains.

Reply:
Well, testing and debugging gets easier if we use one subdomain (and one Vercel project) per chain.  That's how we are doing it with the other Nation3 dApps, so probably makes most sense to treat the Agreements dApp the same way?
```
app.nation3.org, app-sepolia.nation3.org
income.nation3.org, income-sepolia.nation3.org
agreements.nation3.org, agreements-sepolia.nation3.org
```